### ISSUE TITLE
Using l2 bow dc's from server

### ISSUE THREAD

**Issue by schindlershadow** [Minecraft-LightLand/L2Archery#10]
trying to use starter bow dc's player

DC report https://pastebin.com/HrLnArsM

server log:
```
[11:10:41] [Server thread/ERROR] Attempted to load class net/minecraft/client/multiplayer/ClientLevel for invalid dist DEDICATED_SERVER
[11:10:41] [Server thread/ERROR] Failed to handle packet net.minecraft.network.protocol.game.ServerboundUseItemPacket@e144f61, suppressing error
[11:10:41] [Server thread/ERROR] Attempted to load class net/minecraft/client/multiplayer/ClientLevel for invalid dist DEDICATED_SERVER
[11:10:41] [Server thread/WARN] Failed to handle packet for /[REDACTED]:52298
[11:10:41] [Server thread/INFO] SchindlerShadow lost connection: Internal server error
[11:10:41] [Server thread/INFO] SchindlerShadow <Overworld> left the game
[11:10:41] [Server thread/INFO] [voicechat] Disconnecting client SchindlerShadow <Overworld>
```

minecraft@1.21.1
l2archery-3.0.0+8
NeoForge: 21.1.42

**Comment by lcy0x1** [comment]
Send me latest.log

Crash Report doesn't seems helpful

**Comment by schindlershadow** [comment]
client https://pastebin.com/raw/e1yhkRm3
server https://pastebin.com/raw/v4QgW2zf

looks like its this part
```
[01Oct2024 23:49:46.746] [Server thread/ERROR] [net.neoforged.fml.common.asm.RuntimeDistCleaner/DISTXFORM]: Attempted to load class net/minecraft/client/multiplayer/ClientLevel for invalid dist DEDICATED_SERVER
[01Oct2024 23:49:46.747] [Server thread/ERROR] [net.minecraft.network.protocol.game.ServerPacketListener/]: Failed to handle packet net.minecraft.network.protocol.game.ServerboundUseItemPacket@60a0635b, suppressing error
java.lang.RuntimeException: Attempted to load class net/minecraft/client/multiplayer/ClientLevel for invalid dist DEDICATED_SERVER
	at MC-BOOTSTRAP/fml_loader@4.0.24/net.neoforged.fml.common.asm.RuntimeDistCleaner.processClassWithFlags(RuntimeDistCleaner.java:60) ~[loader-4.0.24.jar%2360!/:4.0]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:94) ~[modlauncher-11.0.4.jar%2362!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) ~[modlauncher-11.0.4.jar%2362!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes(TransformingClassLoader.java:57) ~[modlauncher-11.0.4.jar%2362!/:?]
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.readerToClass(ModuleClassLoader.java:190) ~[securejarhandler-3.0.8.jar:?]
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.lambda$findClass$20(ModuleClassLoader.java:298) ~[securejarhandler-3.0.8.jar:?]
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.loadFromModule(ModuleClassLoader.java:311) ~[securejarhandler-3.0.8.jar:?]
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.findClass(ModuleClassLoader.java:298) ~[securejarhandler-3.0.8.jar:?]
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:214) ~[securejarhandler-3.0.8.jar:?]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
	at TRANSFORMER/l2archery@3.0.0+8/dev.xkmc.l2archery.content.item.BowConfig.getEffects(BowConfig.java:23) ~[l2archery-3.0.0+8.jar%23503!/:3.0.0+8]
	at TRANSFORMER/l2archery@3.0.0+8/dev.xkmc.l2archery.content.item.GenericBowItem.getFeatures(GenericBowItem.java:236) ~[l2archery-3.0.0+8.jar%23503!/:3.0.0+8]
	at TRANSFORMER/l2archery@3.0.0+8/dev.xkmc.l2archery.content.controller.BowFeatureController.startUsing(BowFeatureController.java:10) ~[l2archery-3.0.0+8.jar%23503!/:3.0.0+8]
	at TRANSFORMER/l2archery@3.0.0+8/dev.xkmc.l2archery.content.item.GenericBowItem.use(GenericBowItem.java:126) ~[l2archery-3.0.0+8.jar%23503!/:3.0.0+8]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.item.ItemStack.use(ItemStack.java:391) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.level.ServerPlayerGameMode.useItem(ServerPlayerGameMode.java:313) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.network.ServerGamePacketListenerImpl.handleUseItem(ServerGamePacketListenerImpl.java:1171) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.game.ServerboundUseItemPacket.handle(ServerboundUseItemPacket.java:45) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.game.ServerboundUseItemPacket.handle(ServerboundUseItemPacket.java:9) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.protocol.PacketUtils.lambda$ensureRunningOnSameThread$0(PacketUtils.java:27) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.TickTask.run(TickTask.java:18) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.doRunTask(BlockableEventLoop.java:148) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.ReentrantBlockableEventLoop.doRunTask(ReentrantBlockableEventLoop.java:23) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:872) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.doRunTask(MinecraftServer.java:170) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.pollTask(BlockableEventLoop.java:122) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.pollTaskInternal(MinecraftServer.java:855) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.pollTask(MinecraftServer.java:849) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.util.thread.BlockableEventLoop.runAllTasks(BlockableEventLoop.java:111) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.waitUntilNextTick(MinecraftServer.java:825) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:712) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:267) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
[01Oct2024 23:49:46.795] [Server thread/ERROR] [net.neoforged.fml.common.asm.RuntimeDistCleaner/DISTXFORM]: Attempted to load class net/minecraft/client/multiplayer/ClientLevel for invalid dist DEDICATED_SERVER
[01Oct2024 23:49:46.844] [Server thread/WARN] [net.minecraft.server.network.ServerConnectionListener/]: Failed to handle packet for /192.168.2.1:53191
net.minecraft.ReportedException: Ticking player
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:625) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.network.ServerGamePacketListenerImpl.tick(ServerGamePacketListenerImpl.java:260) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.network.Connection.tick(Connection.java:412) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.network.ServerConnectionListener.tick(ServerConnectionListener.java:159) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.tickChildren(MinecraftServer.java:1051) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.dedicated.DedicatedServer.tickChildren(DedicatedServer.java:317) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.tickServer(MinecraftServer.java:917) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:707) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.MinecraftServer.lambda$spin$2(MinecraftServer.java:267) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at java.base/java.lang.Thread.run(Thread.java:1583) [?:?]
Caused by: java.lang.RuntimeException: Attempted to load class net/minecraft/client/multiplayer/ClientLevel for invalid dist DEDICATED_SERVER
	at MC-BOOTSTRAP/fml_loader@4.0.24/net.neoforged.fml.common.asm.RuntimeDistCleaner.processClassWithFlags(RuntimeDistCleaner.java:60) ~[loader-4.0.24.jar%2360!/:4.0]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.LaunchPluginHandler.offerClassNodeToPlugins(LaunchPluginHandler.java:94) ~[modlauncher-11.0.4.jar%2362!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) ~[modlauncher-11.0.4.jar%2362!/:?]
	at MC-BOOTSTRAP/cpw.mods.modlauncher@11.0.4/cpw.mods.modlauncher.TransformingClassLoader.maybeTransformClassBytes(TransformingClassLoader.java:57) ~[modlauncher-11.0.4.jar%2362!/:?]
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.readerToClass(ModuleClassLoader.java:190) ~[securejarhandler-3.0.8.jar:?]
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.lambda$findClass$20(ModuleClassLoader.java:298) ~[securejarhandler-3.0.8.jar:?]
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.loadFromModule(ModuleClassLoader.java:311) ~[securejarhandler-3.0.8.jar:?]
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.findClass(ModuleClassLoader.java:298) ~[securejarhandler-3.0.8.jar:?]
	at cpw.mods.securejarhandler/cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:214) ~[securejarhandler-3.0.8.jar:?]
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:526) ~[?:?]
	at TRANSFORMER/l2archery@3.0.0+8/dev.xkmc.l2archery.content.item.BowConfig.getEffects(BowConfig.java:23) ~[l2archery-3.0.0+8.jar%23503!/:3.0.0+8]
	at TRANSFORMER/l2archery@3.0.0+8/dev.xkmc.l2archery.content.item.GenericBowItem.getFeatures(GenericBowItem.java:236) ~[l2archery-3.0.0+8.jar%23503!/:3.0.0+8]
	at TRANSFORMER/l2archery@3.0.0+8/dev.xkmc.l2archery.content.controller.BowFeatureController.usingTick(BowFeatureController.java:14) ~[l2archery-3.0.0+8.jar%23503!/:3.0.0+8]
	at TRANSFORMER/l2archery@3.0.0+8/dev.xkmc.l2archery.content.item.GenericBowItem.onUseTick(GenericBowItem.java:120) ~[l2archery-3.0.0+8.jar%23503!/:3.0.0+8]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.item.ItemStack.onUseTick(ItemStack.java:1069) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.entity.LivingEntity.updateUsingItem(LivingEntity.java:3078) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.level.ServerPlayer.updateUsingItem(ServerPlayer.java:1980) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.entity.LivingEntity.updatingUsingItem(LivingEntity.java:3067) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.entity.LivingEntity.tick(LivingEntity.java:2393) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.world.entity.player.Player.tick(Player.java:281) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	at TRANSFORMER/minecraft@1.21.1/net.minecraft.server.level.ServerPlayer.doTick(ServerPlayer.java:556) ~[server-1.21.1-20240808.144430-srg.jar%23364!/:?]
	... 9 more
[01Oct2024 23:49:46.856] [Server thread/INFO] [net.minecraft.server.network.ServerGamePacketListenerImpl/]: SchindlerShadow lost connection: Internal server error
[01Oct2024 23:49:46.856] [Server thread/INFO] [net.minecraft.server.MinecraftServer/]: SchindlerShadow <Overworld> left the game
```

strangely its fine in SP

**Comment by lcy0x1** [comment]
thanks this is helpful

**Comment by lcy0x1** [comment]
Fixed in L2Library 3.0.2+7 and L2Archery 3.0.0+9
